alien4cloud-core/src/main/java/alien4cloud/topology/TopologyService.java

```diff
@@ -424,13 +349,56 @@ public void isUniqueNodeTemplateName(Topology topology, String newNodeTemplateNa
         }
     }
 
+    public void updateDependencies(EditionContext context, CSARDependency newDependency) {
```

Comment:
CSARDependency newDependency
=> this should not be a newDependency but only an updatedDependency.

This code should not be called for new dependency as the rollback code (in case of NotFoundException) does not remove the newDependency from the context...

Actually there is probably a bug there as introduced transitive dependencies from newDependency set are never removed from the TOSCA context.